lib/puppet/resource.rb

```diff
@@ -87,17 +98,21 @@ def self.convert_rich_data_hash(rich_data_hash, rich_data_enabled)
       rich_data_hash
     end
   end
-  private_class_method :convert_rich_data_hash
+  private :convert_rich_data_hash
 
   def inspect
     "#{@type}[#{@title}]#{to_hash.inspect}"
   end
 
   def to_data_hash(rich_data_enabled = false)
-    data = ([:type, :title, :tags] + ATTRIBUTES).inject({}) do |hash, param|
-      next hash unless value = self.send(param)
-      hash[param.to_s] = value
-      hash
+    data = {
+      'type' => type,
+      'title' => title,
+      'tags' => tags.to_data_hash
+    }
+    ATTRIBUTES.each do |param|
+      value = send(param)
+      data[param.to_s] = value if value
```

Comment:
Looks like a bug in the old code. If `value` is `false` then we never set the value. Seems like that should be `unless value.nil?`